Utilities/DataLoading/EdFi.LoadTools/ApiClient/XsdMetadata/XsdFilesRetriever.cs

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using EdFi.Common.Extensions;
+using log4net;
+using Microsoft.Extensions.Configuration;
+
+namespace EdFi.LoadTools.ApiClient.XsdMetadata
+{
+    public class XsdFilesRetriever : IXsdFilesRetriever, IDisposable
+    {
+        private const string EdFi = "ed-fi";
+        private readonly IConfiguration _configuration;
+        private readonly string _extension;
+        private readonly HttpClient _httpClient;
+        private readonly ILog _logger = LogManager.GetLogger(typeof(XsdFilesRetriever));
+        private readonly OdsVersionInformation _odsVersionInformation;
+        private readonly IRemoteFileDownloader _remoteFileDownloader;
+        private readonly string _xsdMedataUrl;
+        private readonly IXsdMetadataFilesProvider _xsdMetadataFilesProvider;
+        private readonly IXsdMetadataInformationProvider _xsdMetadataInformationProvider;
+        private string _xsdFolder;
+
+        public XsdFilesRetriever(IConfiguration configuration,
+            IXsdMetadataInformationProvider xsdMetadataInformationProvider,
+            IXsdMetadataFilesProvider xsdMetadataFilesProvider,
+            IRemoteFileDownloader remoteFileDownloader,
+            OdsVersionInformation odsVersionInformation)
+        {
+            _configuration = configuration;
+            _xsdMetadataInformationProvider = xsdMetadataInformationProvider;
+            _xsdMetadataFilesProvider = xsdMetadataFilesProvider;
+            _remoteFileDownloader = remoteFileDownloader;
+            _odsVersionInformation = odsVersionInformation;
+            _httpClient = new HttpClient {Timeout = new TimeSpan(0, 0, 5, 0)};
+
+            _xsdMedataUrl = _configuration.GetValue<string>("OdsApi:XsdMetadataUrl");
+            _extension = _configuration.GetValue<string>("OdsApi:Extension");
+
+            SetXsdFolder();
+
+            void SetXsdFolder()
+            {
+                string currentDirectory = Directory.GetCurrentDirectory();
+
+                string workingFolder = string.IsNullOrEmpty(_configuration.GetValue<string>("Folders:Working"))
+                    ? currentDirectory
+                    : Path.GetFullPath(_configuration.GetValue<string>("Folders:Working"));
+
+                _xsdFolder = string.IsNullOrEmpty(_configuration.GetValue<string>("Folders:Xsd"))
+                    ? Path.Combine(workingFolder, "xsd")
+                    : Path.GetFullPath(_configuration.GetValue<string>("Folders:Xsd"));
```

Comment:
should we be using the BulkLoadClientConfiguration here?
https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS/blob/80656f68392f73043a024ab8dd79d10fc4c22417/Utilities/DataLoading/EdFi.LoadTools/BulkLoadClient/Application/BulkLoadClientConfiguration.cs#L134-L140